assets/src/scripts/components/dailyValueHydrograph/index.spec.js

```diff
@@ -108,32 +108,30 @@ describe('components/dailyValueHydrograph/index', () => {
         });
     });
 
-    it('graph-container should be hidden if no current observations time series is set', () => {
-        let store = configureStore({
-            ...TEST_STATE,
-            observationsState: {}
+    it('Should render the hydrograph container', (done) => {
+        attachToNode(configureStore(), testDiv.node(), {siteno: '1213', timeSeriesId: '12345'});
+        jasmine.Ajax.requests.mostRecent().respondWith({
+            status: 200,
+            response: JSON.stringify(TEST_STATE.observationsData.timeSeries['12345'])
         });
-        attachToNode(store, testDiv.node(), {siteno: '1213', timeSeriesId: '12345'});
 
-        expect(testDiv.select('.graph-container').attr('hidden')).toBe('true');
-    });
-
-    it('graph-container should be hidden if no current observations time series is set', () => {
-        let store = configureStore(TEST_STATE);
-        attachToNode(store, testDiv.node(), {siteno: '1213', timeSeriesId: '12345'});
-
-        expect(testDiv.select('.graph-container').attr('hidden')).toBeNull();
+        window.requestAnimationFrame(() => {
+            expect(testDiv.select('.graph-container').selectAll('.hydrograph-container').size()).toBe(2);
+            done();
+        });
     });
 
-    it('Should render the hydrograph container', () => {
-        attachToNode(configureStore(), testDiv.node(), {siteno: '1213', timeSeriesId: '12345'});
-
-        expect(testDiv.select('.graph-container').selectAll('.hydrograph-container').size()).toBe(1);
-    });
 
-    it('should render the tooltip cursor slider', () => {
+    it('should render the tooltip cursor slider', (done) => {
         attachToNode(configureStore(TEST_STATE), testDiv.node(), {siteno: '1213', timeSeriesId: '12345'});
+        jasmine.Ajax.requests.mostRecent().respondWith({
+            status: 200,
+            response: JSON.stringify(TEST_STATE.observationsData.timeSeries['12345'])
+        });
 
-        expect(testDiv.select('.graph-container').selectAll('.slider-wrapper').size()).toBe(1);
+        window.requestAnimationFrame(() => {
+            expect(testDiv.select('.graph-container').selectAll('.slider-wrapper').size()).toBe(1);
+            done();
+        });
     });
-});
\ No newline at end of file
+});
```

Comment:
Could you add a test to make sure the legend is rendered? You can use the test for rendering the tooltip cursor slider as an example.